qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -229,7 +229,8 @@ def __init__(
             tuple[str, complex],
             list[tuple[str, complex]],
             list[tuple[str, float]],
-            Sequence[tuple[Iterable[tuple[str, int]], complex]],
+            list[tuple[list[tuple[str, int]], complex]],
+            list[tuple[tuple[tuple[str, int], ...], complex]],
```

Comment:
To be clear: this change is not breaking because the `Sequence` variant was not released yet. So compared to stable the difference is merely adding:
```
            list[tuple[tuple[tuple[str, int], ...], complex]],
```

Reply:
I just want to check this. Other than that this looks straight forward to me :+1:

Reply:
This is not breaking change because this is not released.

Reply:
This is a bit of a discussion, but this part was changed from list to tuple in #607. In that sense, list would be the old API, so should we deprecate it? (In other words, Iterable was introduced as a general expression that included list and tuple, but in real it is not that broad, so I explicitly reverted in this PR.)

Reply:
What would be the benefit of explicitly deprecated the list-case?

Reply:
Well, I guess it's nothing special, just a simplification of the code.

Reply:
Then let's keep it.